### PR TITLE
Add Clock In

### PR DESCRIPTION
**Pull request by jorgecod**
[Issue details](https://github.com/ioet/time-tracker-ui/issues/4)

- When mouse over a project, a clock-in button should appear next to the project name.
- When the clock in button is clicked, show the following fields: Activity, Jira ticket, Technology, and Notes.
- Clock out button should be visible after clock in.

### CODE REVIEW

File: src/app/components/shared/details-fields/details-fields.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { DetailsFieldsComponent } from './details-fields.component';
+
+describe('DetailsFieldsComponent', () => {
+  let component: DetailsFieldsComponent;
+  let fixture: ComponentFixture<DetailsFieldsComponent>;
+
+  beforeEach(async(() => {
```
Comment (enriquezrene) [inline src/app/components/shared/details-fields/details-fields.component.spec.ts]: should not we have a test checking the details fields appear when a project is selected.
what about another test checking the clock-in button appears when the mouse is over a project?

File: src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts [hunk 2]
```diff
@@ -19,7 +18,14 @@ describe('TimeClockComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should be created', () => {
+  it("should be created", () => {
     expect(component).toBeTruthy();
   });
+
+  it("should set showfileds as true", () => {
```
Comment (enriquezrene) [inline src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts]: typo: should set showFields as true

File: src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts [hunk 3]
```diff
@@ -19,7 +18,14 @@ describe('TimeClockComponent', () => {
...
+    const show = true;
+    component.showFields = show;
+    component.getShowFields(show);
```
Comment (enriquezrene) [inline src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts]: It seems this function is changing showFields value. In this case, the field name should be:

setShowFields(show)

File: src/app/components/options-sidebar/time-clock/time-clock.component.ts [hunk 4]
```diff
@@ -1,15 +1,25 @@
...
-  constructor() { }
+  showFields: boolean;
 
-  ngOnInit(): void {
-  }
+  constructor() {}
+
+  ngOnInit(): void {}
 
+  getShowFields(show: boolean) {
```
Comment (enriquezrene) [inline src/app/components/options-sidebar/time-clock/time-clock.component.ts]: definitely, the function name should be: `setShowFields`

File: src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts (hunk 2 above)
Comment (jorgecod) [reply src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts]: Done 👌

File: src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts (hunk 3 above)
Comment (jorgecod) [reply src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts]: Done 👌

File: src/app/components/options-sidebar/time-clock/time-clock.component.ts (hunk 4 above)
Comment (jorgecod) [reply src/app/components/options-sidebar/time-clock/time-clock.component.ts]: Done 👌

File: src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts [hunk 5]
```diff
@@ -19,7 +18,14 @@ describe('TimeClockComponent', () => {
...
 
-  it('should be created', () => {
+  it("should be created", () => {
     expect(component).toBeTruthy();
   });
+
+  it("should set showFields as true", () => {
+    const show = true;
+    component.showFields = show;
+    component.setShowFields(show);
```
Comment (enriquezrene) [inline src/app/components/options-sidebar/time-clock/time-clock.component.spec.ts]: Why are we setting this field to true twice?

**Pull request merged**